astropy/coordinates/matching.py

```diff
@@ -226,6 +226,14 @@ def search_around_3d(coords1, coords2, distlimit, storekdtree='_kdtree_3d'):
                          '``coord1.separation_3d(coord2) < distlimit`` to find '
                          'the coordinates near a scalar coordinate.')
 
+    if len(coords1) == 0 or len(coords2) == 0:
+        # Empty array input: return empty match
+        idxs1 = np.array([], dtype=np.int)
+        idxs2 = np.array([], dtype=np.int)
+        return (np.array([], dtype=np.int), np.array([], dtype=np.int),
+                u.Quantity([], u.deg),
+                u.Quantity([], u.dimensionless_unscaled))
```

Comment:
I see below that an empty `idxs1` leads to a dimensionless return (L261), but I don't understand why -- if the 2d separation is returned as an empty quantity with degree units, why is the 3d separation assumed to be dimensionless unscaled? Maybe @eteq can explain?